### PR TITLE
Make metrics server configurable

### PR DESCRIPTION
**Pull request by zonque**
Add a block in the config file to allow the configuration of host, port and mTLS details for the metrics server.

Luckily we can reuse a lot of the already existing logic in the code base, but some code a new package named `tlsconn`.

The `readContent*` helpers were replaced by `os.ReadFile` from the standard library.

### CODE REVIEW

**Comment by zonque** [comment]
This is currently based on #4. Will rebase once the other one is merged.

**Comment by zonque** [comment]
> This is currently based on #4. Will rebase once the other one is merged.

Rebased now.

**Comment by nothinux** [comment]
Let's enable the metrics by default and provide the user to change the config and disable the metrics, you can add `enabled` configuration in metrics.

```
metrics:
  enabled: false
```

File: pkg/tlsconn/tlsconn.go [hunk 1]
```diff
@@ -96,7 +95,7 @@ func getTLSConfig(c config.TLSConfig) (*ProxyTLS, error) {
 }
 
 func getCACertPool(c config.TLSConfig) (*x509.CertPool, error) {
-	cacert, err := readContent(c.CaCert)
+	cacert, err := ioutil.ReadFile(c.CaCert)
```
[the remark this answers is not in the document]
Comment (zonque) [reply pkg/tlsconn/tlsconn.go]: Done!

**Comment by zonque** [comment]
> Let's enable the metrics by default and provide the user to change the config and disable the metrics, you can add `enabled` configuration in metrics.

I don't think that's a good idea. Configurations should always be explicit, and opt-out directives to disable defaults are odd. This entire tool is all about networking, binding ports etc, and allocating a port that isn't mentioned in the config file is quite confusing.

Besides, it might even be problematic to expose the metrics by default. If the proxy is deployed on a public server, you don't want to give away process internals.

Note that the default when running without a config is unchanged, and users can easily retain the old behavior by adding 3 lines in the config.

**Comment by nothinux** [comment]
> Besides, it might even be problematic to expose the metrics by default. If the proxy is deployed on a public server, you don't want to give away process internals.

That makes sense. yeah, I just realized users can use this proxy as a public-facing service.
the code looks good, will merge this

**Pull request merged**